mypy/newsemanal/semanal.py

```diff
@@ -1915,6 +1915,14 @@ def analyze_namedtuple_assign(self, s: AssignmentStmt) -> bool:
         # Yes, it's a valid namedtuple, but defer if it is not ready.
         if not info:
             self.mark_incomplete(name, lvalue, becomes_typeinfo=True)
+        else:
+            # TODO: This is needed for one-to-one compatibility with old analyzer, otherwise
+            # type checker will try to infer Any for the l.h.s.
+            # Remove this after new analyzer is the default one!
```

Comment:
This comment is a bit unclear. What would happen if we'd infer `Any`? Can we just remove this afterwards, or does something else need to be changed as well?

Reply:
This will cause named tuple class object to be inferred as `Any` in runtime context. This is actually also related to https://github.com/python/mypy/issues/6458

I will update the comment.

Reply:
Just to give a bit more details: since `visit_namedtuple_expr()` returns `AnyType` in type checker, the binder will assign this to the `lvalue` (by its pseudo-hash), so despite the fact that in the symbol table this name points to the `TypeInfo`, it will be "narrowed down" to `Any`.